### PR TITLE
Grayout

### PR DESCRIPTION
**Pull request by jalons**
Persist grayout and pd_disabled across restarts.  Introduces a node in zookeeper that maintains this state.  Known issues are pd_disabled doesn't actually flow through manual_update on disable (it sends an HTTP DELETE instead of posting to manual_update), and tests are broken as I have not researched mox.

### CODE REVIEW

**Comment by jalons** [comment]
@nicmoski

File: server/zoom/www/cache/application_state_cache.py [hunk 1]
```diff
@@ -62,14 +62,54 @@ def _load(self):
...
         """
         Manual override from client of specific value
         :type path: str
         :param key: str
         """
+        # Set the permanent storage
+
         state = self._cache._application_states.get(path, None)
         if state is not None:
+            if key in ['grayed', 'pd_disabled']:
```
Comment (nicmoski) [inline server/zoom/www/cache/application_state_cache.py]: Do we have to hard-code this? If we add some new value we'll have to touch this later. Are we positive we only want to persist 'grated' and 'pd_disabled'?

File: server/zoom/www/cache/application_state_cache.py [hunk 2]
```diff
@@ -250,10 +294,28 @@ def _get_existing_attribute(self, path, attr, default=False):
...
+        
         existing_obj = self._cache._application_states.get(path, None)
+        _override = {}
+
+        try:
+            _data, _unused = self._zoo_keeper.get(self._configuration.override_node)
+            # I don't like grabbing the entire override object, but what can you do?
+            _override = json.loads(_data)
+            grayed_out = _override.get('grayed')
+            pd_disabled = _override.get('pd_disabled')
```
Comment (nicmoski) [inline server/zoom/www/cache/application_state_cache.py]: these variables are not used

File: server/zoom/www/cache/application_state_cache.py (hunk 1 above)
Comment (jalons) [inline server/zoom/www/cache/application_state_cache.py]: No, I don't suppose it's necessary they'd be only ones persisted, and abstracting them is wise.  Let me look.

**Pull request closed without merging**